DQM/Physics/src/SingleTopTChannelLeptonDQM_miniAOD.h

```diff
@@ -0,0 +1,292 @@
+                     ->second->getBinContent(idx + 1);
+      double value = hists_.find((channel + "Eff_").c_str())
+                         ->second->getBinContent(idx + 1);
+      fill(
+          (channel + "Eff_").c_str(), idx + 0.5,
+          1. / evts * (accept(event, triggerTable, selectionPath(labels[idx])) -
+                       value));
+    }
+  }
+}
+}
+
+#include <utility>
+
+#include "DQM/Physics/interface/TopDQMHelpers.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "FWCore/Common/interface/TriggerNames.h"
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/Common/interface/TriggerResults.h"
+#include "DQMServices/Core/interface/DQMEDAnalyzer.h"
+
+
+
+class SingleTopTChannelLeptonDQM_miniAOD : public DQMEDAnalyzer {
+ public:
+  /// default constructor
+  SingleTopTChannelLeptonDQM_miniAOD(const edm::ParameterSet& cfg);
+  /// default destructor
+  ~SingleTopTChannelLeptonDQM_miniAOD() {};
+
+  /// do this during the event loop
+  virtual void analyze(const edm::Event& event, const edm::EventSetup& setup);
+ 
+ protected:
+  //Book histograms
+  void bookHistograms(DQMStore::IBooker &,
+    edm::Run const &, edm::EventSetup const &) override;
+
+ private:
+  /// deduce object type from ParameterSet label, the label
+  /// is expected to be of type 'objectType:selectionStep'
+  std::string objectType(const std::string& label) {
+    return label.substr(0, label.find(':'));
+  };
+  /// deduce selection step from ParameterSet label, the
+  /// label is expected to be of type 'objectType:selectionStep'
+  std::string selectionStep(const std::string& label) {
+    return label.substr(label.find(':') + 1);
+  };
+
+ private:
+  /// trigger table
+  edm::EDGetTokenT<edm::TriggerResults> triggerTable__;
+
+  /// trigger paths
+  std::vector<std::string> triggerPaths_;
+  /// primary vertex
+  edm::InputTag vertex_;
+  edm::EDGetTokenT<reco::Vertex> vertex__;
+  /// string cut selector
+  std::unique_ptr<StringCutObjectSelector<reco::Vertex> > vertexSelect_;
+
+  /// beamspot
+  edm::InputTag beamspot_;
+  edm::EDGetTokenT<reco::BeamSpot> beamspot__;
+  /// string cut selector
+  std::unique_ptr<StringCutObjectSelector<reco::BeamSpot> > beamspotSelect_;
+
+
+  std::vector<std::string> selectionOrder_;
+
+  std::map<
+      std::string,
+      std::pair<edm::ParameterSet,
+                std::unique_ptr<SingleTopTChannelLepton_miniAOD::MonitorEnsemble> > >
+      selection_;
+
+  std::unique_ptr<SelectionStep<pat::Muon> > MuonStep;
+  std::unique_ptr<SelectionStep<pat::Electron> > ElectronStep;
+  std::unique_ptr<SelectionStep<reco::Vertex> > PvStep;
+
+  std::vector<std::unique_ptr<SelectionStep<pat::Jet> > > JetSteps;
+
+  std::unique_ptr<SelectionStep<pat::MET> > METStep;
+  std::vector<edm::ParameterSet> sel;
```

Comment:
please follow a convention for member data names. We prefer a single trailing underscore, and a lower case letter at the start (e.g., muonStep_, metStep_, etc)